geometric FBA does not work with cplex or Gurobi

#### Problem description

Just running the tests with either CPLEX or Gurobi will raise errors in Geometric FBA. The problem is that the code call `get_solution` after exiting the with block which will never work with cplex or gurobi since the solution is invalidated in those solver upon any change to the model (which the context exit triggers).

#### Code Sample
Just run the cobrapy tests with Gurobi or CPLEx installed.

#### Actual Output

```self = <cobra.test.test_flux_analysis.TestCobraFluxAnalysis object at 0x7f062eb2dcc0>, solver = 'optlang-gurobi'

    @pytest.mark.parametrize("solver", all_solvers)
    def test_geometric_fba(self, solver):
        model = construct_geometric_fba_model()
        model.solver = solver
>       geometric_fba_sol = geometric_fba(model)

cobra/test/test_flux_analysis.py:206: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
cobra/flux_analysis/geometric.py:106: in geometric_fba
    return get_solution(model)
cobra/core/solution.py:297: in get_solution
    var_primals = model.solver.primal_values
../../.local/lib/python3.6/site-packages/optlang/interface.py:1234: in primal_values
    zip(self._get_variables_names(), self._get_primal_values())
../../.local/lib/python3.6/site-packages/optlang/interface.py:1245: in _get_primal_values
    return [variable.primal for variable in self.variables]
../../.local/lib/python3.6/site-packages/optlang/interface.py:1245: in <listcomp>
    return [variable.primal for variable in self.variables]
../../.local/lib/python3.6/site-packages/optlang/interface.py:273: in primal
    return self._get_primal()
../../.local/lib/python3.6/site-packages/optlang/gurobi_interface.py:158: in _get_primal
    return self._internal_variable.getAttr('X')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   AttributeError: b"Unable to retrieve attribute 'X'"

var.pxi:142: AttributeError
```

#### Expected Output

Tests should pass.

#### Dependency Information

Please paste the output of `python -c "from cobra import show_versions;
show_versions()"` in the details block below.

<details>

System Information
==================
OS                     Linux
OS-release 4.15.0-22-generic
Python                 3.6.5

Package Versions
================
cobra        0.12.0
depinfo       1.3.0
future       0.16.0
numpy        1.14.3
optlang       1.4.2
pandas       0.23.0
pip          10.0.1
ruamel.yaml 0.14.12
setuptools   39.2.0
six          1.11.0
swiglpk       1.4.4
tabulate      0.8.2
wheel        0.31.1


</details>

--- a/cobra/flux_analysis/geometric.py
+++ b/cobra/flux_analysis/geometric.py
@@ -79,7 +79,7 @@ def geometric_fba(model, epsilon=1E-06, max_tries=200):
         # minimize distance between flux and centre
         model.objective = prob.Objective(Zero, sloppy=True, direction="min")
         model.objective.set_linear_coefficients({v: 1.0 for v in obj_vars})
-        model.optimize()
+        sol = model.optimize()
 
         # further iterations
         while delta > epsilon and count <= max_tries:
@@ -90,7 +90,7 @@ def geometric_fba(model, epsilon=1E-06, max_tries=200):
                 var.ub = mean_flux[rxn_id]
                 u_c.ub = mean_flux[rxn_id]
                 l_c.lb = fva_sol.at[rxn_id, "minimum"]
-            model.optimize()
+            sol = model.optimize()
             delta = (fva_sol["maximum"] - fva_sol["minimum"]).max()
 
             count += 1
@@ -103,4 +103,4 @@ def geometric_fba(model, epsilon=1E-06, max_tries=200):
                     "maximum iterations".format(max_tries)
                 )
 
-    return get_solution(model)
+    return sol

--- a/cobra/test/test_flux_analysis.py
+++ b/cobra/test/test_flux_analysis.py
@@ -261,14 +261,14 @@ class TestCobraFluxAnalysis:
             ssq = (knock_sol.fluxes - sol.fluxes).pow(2).sum()
 
         with model:
-            add_moma(model)
+            add_moma(model, linear=False)
             model.reactions.PFK.knock_out()
             moma_sol = model.optimize()
             moma_ssq = (moma_sol.fluxes - sol.fluxes).pow(2).sum()
 
         # Use normal FBA as reference solution.
         with model:
-            add_moma(model, solution=sol)
+            add_moma(model, solution=sol, linear=False)
             model.reactions.PFK.knock_out()
             moma_ref_sol = model.optimize()
             moma_ref_ssq = (moma_ref_sol.fluxes - sol.fluxes).pow(2).sum()

